### DISCUSSION TITLE
Converting nb::ndarray to PyObject *

### DISCUSSION THREAD

**Discussion by ncullen93** [wjakob/nanobind#522] [General]
Hey there! I was wondering if anyone knows how to convert a nb::ndarray to a PyObject * so that it can be passed into the `PyObject_GetBuffer` function, for instance. I've tried calling `.data()` and casting to a PyObject * but i'm getting lots of segfaults. 

In pybind11, you could simply call `array.ptr()` but that only seems possible in nanobind now for things like `nb::tuple` that have a `nb::handle` ?. I guess I would have to get the owner or  somehow do the reverse of `nb::handle`... 

I think this might be related to https://github.com/wjakob/nanobind/discussions/519 but unsure

To be more specific, I am trying to take in a numpy array and create an ITK image from it. Something like this:

```c++
template <typename ImageType>
void fromNumpy( nb::ndarray<nb::numpy> data, nb::tuple datashape )
{
    typedef typename ImageType::Pointer ImagePointerType;
    ImagePointerType myimage = ImageType::New();
    typedef itk::PyBuffer<ImageType> PyBufferType;
   
   // data.ptr() doesnt exist anymore
  PyObject * arrData = .... 

    myimage = PyBufferType::_GetImageViewFromArray( arrData , datashape.ptr());
}
```

**Comment by wjakob** [comment]
``nb::object o = nb::cast(data);`` and then use `o.ptr()`

**Reply by ncullen93** [reply]
Thanks I had JUST come upon nb::find and nb::cast from https://nanobind.readthedocs.io/en/latest/why.html#minor-additions and was going to try it. Much appreciated.
